Sources/Testing/Test.Case.Generator.swift

```diff
@@ -72,15 +72,39 @@ extension Test.Case {
     /// - Parameters:
     ///   - collection: The collection of argument values for which test cases
     ///     should be generated.
+    ///   - parameters: The parameters of the test function for which test cases
+    ///     should be generated.
     ///   - testFunction: The test function to which each generated test case
     ///     passes an argument value from `collection`.
+    ///
+    /// This initializer is disfavored since it relies on `Mirror` to
+    /// de-structure elements of tuples. Other initializers which are
+    /// specialized to handle collections of tuple types more efficiently should
+    /// be preferred.
+    @_disfavoredOverload
     init(
       arguments collection: S,
+      parameters: [Test.ParameterInfo],
       testFunction: @escaping @Sendable (S.Element) async throws -> Void
     ) where S: Collection {
-      self.init(sequence: collection) { element in
-        Test.Case(arguments: [element]) {
-          try await testFunction(element)
+      if parameters.count > 1 {
```

Comment:
This change is going to cause us to front-load mirroring rather than performing it lazily, which may have a negative impact on performance when the argument to a test is a sequence of tuples (but not a dictionary). Are we sure we are okay with taking that hit?

Reply:
The other `init` overload I added below, the one for `Collection` values whose `Element` are 2-tuples, is meant to handle that. That means this overload (which now has `@_disfavoredOverload`) will only be used for collections of tuples of 3+ elements. With some further variadic generic improvements which are [in progress](https://github.com/apple/swift/pull/67594), I believe we may be able to solve this for all tuple lengths. 

So yeah I do think this is justified, and the specialized generic overloads help minimize its impact in common cases.